tachyon/zk/base/entity.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_BASE_ENTITY_H_
+#define TACHYON_ZK_BASE_ENTITY_H_
+
+#include <memory>
+#include <utility>
+
+#include "tachyon/zk/transcript/transcript.h"
+
+namespace tachyon::zk {
+
```

Comment:
I think it might be helpful to add some simple comments that the `Entity` is the parent class that encompasses both prover and verifier.